File: 백준_실버/바이러스.py
# 2606

def dfs(now_node):
    if visited[now_node] == True:
        return
    visited[now_node] = True
    for next_node in graph[now_node]:
        dfs(next_node)
    return


if __name__ == "__main__":
    N = int(input())
    graph = [[] for _ in range(N+1)]
    visited = [False] * (N+1)
    
    # 간선의 개수
    linked = int(input())
    
    for _ in range(linked):
        v1, v2 = map(int,input().split())
        graph[v1].append(v2)
        graph[v2].append(v1)
    
    dfs(1)
    
    # 1번 컴퓨터를 제외한 네트워크에 포함된 컴퓨터 개수 출력
    print(visited.count(True) - 1)

File: 백준_실버/바이러스.js
// 2606

const file = process.platform === "linux" ? "dev/stdin" : "./input.txt";
const input = require('fs').readFileSync(file).toString().trim().split('\n');

const N = parseInt(input[0].trim());

const graph = Array.from(Array(N + 1), () => new Array());
const visited = new Array(N + 1).fill(false);

for (let i = 2; i < input.length; i++) {
    let [v1, v2] = input[i].trim().split(' ').map(Number);
    graph[v1].push(v2);
    graph[v2].push(v1);
}

function bfs() {
    const queue = [];

    let cnt = 0;

    queue.push(1);
    visited[1] = true;

    while (queue.length > 0) {
        const now_v = queue.shift();

        graph[now_v].forEach(next_v => {
            if (!visited[next_v]) {
                visited[next_v] = true;
                queue.push(next_v);
                cnt++;
            }
        });
    }

    return cnt;
}

console.log(bfs());



